### PR TITLE
spyder: 5.4.2 -> 5.4.5

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Update spyder to the latest version.
Changelog: https://github.com/spyder-ide/spyder/releases/tag/v5.4.5

cc @gebner 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 253252` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>129 packages built:</summary>
  <ul>
    <li>corrscope</li>
    <li>corrscope.dist</li>
    <li>electron-cash</li>
    <li>electron-cash.dist</li>
    <li>electrum</li>
    <li>electrum-grs</li>
    <li>electrum-grs.dist</li>
    <li>electrum-ltc</li>
    <li>electrum-ltc.dist</li>
    <li>electrum.dist</li>
    <li>flent</li>
    <li>flent.dist</li>
    <li>git-cola</li>
    <li>git-cola.dist</li>
    <li>hydrus</li>
    <li>hydrus.doc</li>
    <li>ihaskell</li>
    <li>inkcut</li>
    <li>inkcut.dist</li>
    <li>inkscape-extensions.inkcut</li>
    <li>jellyfin-mpv-shim</li>
    <li>jellyfin-mpv-shim.dist</li>
    <li>napari (python310Packages.napari)</li>
    <li>napari.dist (python310Packages.napari.dist)</li>
    <li>natron</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.ansible-kernel.dist</li>
    <li>python310Packages.augmax</li>
    <li>python310Packages.augmax.dist</li>
    <li>python310Packages.einops</li>
    <li>python310Packages.einops.dist</li>
    <li>python310Packages.enaml</li>
    <li>python310Packages.enaml.dist</li>
    <li>python310Packages.enamlx</li>
    <li>python310Packages.enamlx.dist</li>
    <li>python310Packages.encodec</li>
    <li>python310Packages.encodec.dist</li>
    <li>python310Packages.jupyter</li>
    <li>python310Packages.jupyter.dist</li>
    <li>python310Packages.magicgui</li>
    <li>python310Packages.magicgui.dist</li>
    <li>python310Packages.napari-console</li>
    <li>python310Packages.napari-console.dist</li>
    <li>python310Packages.napari-npe2</li>
    <li>python310Packages.napari-npe2.dist</li>
    <li>python310Packages.nbdev</li>
    <li>python310Packages.nbdev.dist</li>
    <li>python310Packages.orange3</li>
    <li>python310Packages.orange3.dist</li>
    <li>python310Packages.phik</li>
    <li>python310Packages.phik.dist</li>
    <li>python310Packages.pydeck</li>
    <li>python310Packages.pydeck.dist</li>
    <li>python310Packages.pywebview</li>
    <li>python310Packages.pywebview.dist</li>
    <li>python310Packages.qdarkstyle</li>
    <li>python310Packages.qdarkstyle.dist</li>
    <li>python310Packages.qreactor</li>
    <li>python310Packages.qreactor.dist</li>
    <li>python310Packages.qtawesome</li>
    <li>python310Packages.qtawesome.dist</li>
    <li>python310Packages.qtconsole</li>
    <li>python310Packages.qtconsole.dist</li>
    <li>python310Packages.qtpy</li>
    <li>python310Packages.qtpy.dist</li>
    <li>spyder (python310Packages.spyder)</li>
    <li>python310Packages.spyder-kernels</li>
    <li>python310Packages.spyder-kernels.dist</li>
    <li>spyder.dist (python310Packages.spyder.dist)</li>
    <li>streamlit (python310Packages.streamlit)</li>
    <li>streamlit.dist (python310Packages.streamlit.dist)</li>
    <li>python310Packages.stytra</li>
    <li>python310Packages.stytra.dist</li>
    <li>python310Packages.superqt</li>
    <li>python310Packages.superqt.dist</li>
    <li>python310Packages.vega</li>
    <li>python310Packages.vega.dist</li>
    <li>python310Packages.xarray-einstats</li>
    <li>python310Packages.xarray-einstats.dist</li>
    <li>python311Packages.ansible-kernel</li>
    <li>python311Packages.ansible-kernel.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.einops</li>
    <li>python311Packages.einops.dist</li>
    <li>python311Packages.enaml</li>
    <li>python311Packages.enaml.dist</li>
    <li>python311Packages.enamlx</li>
    <li>python311Packages.enamlx.dist</li>
    <li>python311Packages.encodec</li>
    <li>python311Packages.encodec.dist</li>
    <li>python311Packages.jupyter</li>
    <li>python311Packages.jupyter.dist</li>
    <li>python311Packages.napari-console</li>
    <li>python311Packages.napari-console.dist</li>
    <li>python311Packages.nbdev</li>
    <li>python311Packages.nbdev.dist</li>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python311Packages.pydeck</li>
    <li>python311Packages.pydeck.dist</li>
    <li>python311Packages.pywebview</li>
    <li>python311Packages.pywebview.dist</li>
    <li>python311Packages.qdarkstyle</li>
    <li>python311Packages.qdarkstyle.dist</li>
    <li>python311Packages.qreactor</li>
    <li>python311Packages.qreactor.dist</li>
    <li>python311Packages.qtawesome</li>
    <li>python311Packages.qtawesome.dist</li>
    <li>python311Packages.qtconsole</li>
    <li>python311Packages.qtconsole.dist</li>
    <li>python311Packages.qtpy</li>
    <li>python311Packages.qtpy.dist</li>
    <li>python311Packages.spyder</li>
    <li>python311Packages.spyder-kernels</li>
    <li>python311Packages.spyder-kernels.dist</li>
    <li>python311Packages.spyder.dist</li>
    <li>python311Packages.streamlit</li>
    <li>python311Packages.streamlit.dist</li>
    <li>python311Packages.superqt</li>
    <li>python311Packages.superqt.dist</li>
    <li>python311Packages.vega</li>
    <li>python311Packages.vega.dist</li>
    <li>quarto</li>
    <li>rare</li>
    <li>rare.dist</li>
    <li>smb3-foundry</li>
    <li>vorta</li>
    <li>vorta.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/spyder/default.nix [hunk 1]
```diff
@@ -41,24 +37,21 @@
...
   disabled = pythonOlder "3.7";
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-L8zgT7M7N+P5/9FQgf6ab7afUQXC1afzwUjAp6yKxC8=";
+    hash = "sha256-/9p/8avjy1c3Dwos9Byx03kfVrRofVQus+Ae5beFnmo=";
   };
 
-  patches = [
-    ./dont-clear-pythonpath.patch
```
Comment (bjornfor) [inline pkgs/development/python-modules/spyder/default.nix:59]: Was this merged upstream/not needed anymore?

Comment (GaetanLepage) [reply pkgs/development/python-modules/spyder/default.nix]: Actually it is still relevant, my mistake.
I just added it back.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 253252` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>129 packages built:</summary>
  <ul>
    <li>corrscope</li>
    <li>corrscope.dist</li>
    <li>electron-cash</li>
    <li>electron-cash.dist</li>
    <li>electrum</li>
    <li>electrum-grs</li>
    <li>electrum-grs.dist</li>
    <li>electrum-ltc</li>
    <li>electrum-ltc.dist</li>
    <li>electrum.dist</li>
    <li>flent</li>
    <li>flent.dist</li>
    <li>git-cola</li>
    <li>git-cola.dist</li>
    <li>hydrus</li>
    <li>hydrus.doc</li>
    <li>ihaskell</li>
    <li>inkcut</li>
    <li>inkcut.dist</li>
    <li>inkscape-extensions.inkcut</li>
    <li>jellyfin-mpv-shim</li>
    <li>jellyfin-mpv-shim.dist</li>
    <li>napari (python310Packages.napari)</li>
    <li>napari.dist (python310Packages.napari.dist)</li>
    <li>natron</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.ansible-kernel.dist</li>
    <li>python310Packages.augmax</li>
    <li>python310Packages.augmax.dist</li>
    <li>python310Packages.einops</li>
    <li>python310Packages.einops.dist</li>
    <li>python310Packages.enaml</li>
    <li>python310Packages.enaml.dist</li>
    <li>python310Packages.enamlx</li>
    <li>python310Packages.enamlx.dist</li>
    <li>python310Packages.encodec</li>
    <li>python310Packages.encodec.dist</li>
    <li>python310Packages.jupyter</li>
    <li>python310Packages.jupyter.dist</li>
    <li>python310Packages.magicgui</li>
    <li>python310Packages.magicgui.dist</li>
    <li>python310Packages.napari-console</li>
    <li>python310Packages.napari-console.dist</li>
    <li>python310Packages.napari-npe2</li>
    <li>python310Packages.napari-npe2.dist</li>
    <li>python310Packages.nbdev</li>
    <li>python310Packages.nbdev.dist</li>
    <li>python310Packages.orange3</li>
    <li>python310Packages.orange3.dist</li>
    <li>python310Packages.phik</li>
    <li>python310Packages.phik.dist</li>
    <li>python310Packages.pydeck</li>
    <li>python310Packages.pydeck.dist</li>
    <li>python310Packages.pywebview</li>
    <li>python310Packages.pywebview.dist</li>
    <li>python310Packages.qdarkstyle</li>
    <li>python310Packages.qdarkstyle.dist</li>
    <li>python310Packages.qreactor</li>
    <li>python310Packages.qreactor.dist</li>
    <li>python310Packages.qtawesome</li>
    <li>python310Packages.qtawesome.dist</li>
    <li>python310Packages.qtconsole</li>
    <li>python310Packages.qtconsole.dist</li>
    <li>python310Packages.qtpy</li>
    <li>python310Packages.qtpy.dist</li>
    <li>spyder (python310Packages.spyder)</li>
    <li>python310Packages.spyder-kernels</li>
    <li>python310Packages.spyder-kernels.dist</li>
    <li>spyder.dist (python310Packages.spyder.dist)</li>
    <li>streamlit (python310Packages.streamlit)</li>
    <li>streamlit.dist (python310Packages.streamlit.dist)</li>
    <li>python310Packages.stytra</li>
    <li>python310Packages.stytra.dist</li>
    <li>python310Packages.superqt</li>
    <li>python310Packages.superqt.dist</li>
    <li>python310Packages.vega</li>
    <li>python310Packages.vega.dist</li>
    <li>python310Packages.xarray-einstats</li>
    <li>python310Packages.xarray-einstats.dist</li>
    <li>python311Packages.ansible-kernel</li>
    <li>python311Packages.ansible-kernel.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.einops</li>
    <li>python311Packages.einops.dist</li>
    <li>python311Packages.enaml</li>
    <li>python311Packages.enaml.dist</li>
    <li>python311Packages.enamlx</li>
    <li>python311Packages.enamlx.dist</li>
    <li>python311Packages.encodec</li>
    <li>python311Packages.encodec.dist</li>
    <li>python311Packages.jupyter</li>
    <li>python311Packages.jupyter.dist</li>
    <li>python311Packages.napari-console</li>
    <li>python311Packages.napari-console.dist</li>
    <li>python311Packages.nbdev</li>
    <li>python311Packages.nbdev.dist</li>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python311Packages.pydeck</li>
    <li>python311Packages.pydeck.dist</li>
    <li>python311Packages.pywebview</li>
    <li>python311Packages.pywebview.dist</li>
    <li>python311Packages.qdarkstyle</li>
    <li>python311Packages.qdarkstyle.dist</li>
    <li>python311Packages.qreactor</li>
    <li>python311Packages.qreactor.dist</li>
    <li>python311Packages.qtawesome</li>
    <li>python311Packages.qtawesome.dist</li>
    <li>python311Packages.qtconsole</li>
    <li>python311Packages.qtconsole.dist</li>
    <li>python311Packages.qtpy</li>
    <li>python311Packages.qtpy.dist</li>
    <li>python311Packages.spyder</li>
    <li>python311Packages.spyder-kernels</li>
    <li>python311Packages.spyder-kernels.dist</li>
    <li>python311Packages.spyder.dist</li>
    <li>python311Packages.streamlit</li>
    <li>python311Packages.streamlit.dist</li>
    <li>python311Packages.superqt</li>
    <li>python311Packages.superqt.dist</li>
    <li>python311Packages.vega</li>
    <li>python311Packages.vega.dist</li>
    <li>quarto</li>
    <li>rare</li>
    <li>rare.dist</li>
    <li>smb3-foundry</li>
    <li>vorta</li>
    <li>vorta.dist</li>
  </ul>
</details>

**Review by bjornfor — APPROVED** [review]

**Pull request merged**